gno.land/pkg/gnoland/param.go

```diff
@@ -0,0 +1,121 @@
+package gnoland
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+	"github.com/gnolang/gno/tm2/pkg/sdk/params"
+)
+
+type Param struct {
```

Comment:
This seems to tick all the checkboxes of an interface and not a concrete type, even for the custom Amino marshallers. What's the reasoning behind keeping it a concrete type? I'm assuming simplicity

Reply:
I believe the main reason is that one of the expected usages is to predefine a `var p Param` and then later call `p.Parse(str)`. I'm not particularly attached to this pattern, but it helps maintain alignment with `x/params` from the Cosmos SDK.

We could switch to an interface with five struct implementations. How much do you think this would improve the situation?

Reply:
I don't think there are drastic benefits, as you would need to centralize the `Parse`, which would return a `Param` interface (with an underlying implementation). I was just curious about the thinking